static/js/src/ua-product-selector.js

```diff
@@ -0,0 +1,165 @@
+
+    switch (step) {
+      case "type":
+        quantityTypeEl.innerHTML = `${inputElement.dataset.productName}s`;
+        updateProductState(inputElement);
+        break;
+      case "quantity":
+        if (validQuantity) {
+          updateProductState(inputElement);
+        } else {
+          productState.reset("quantity");
+        }
+        break;
+      default:
+        updateProductState(inputElement);
+    }
+  }
+
+  function render() {
+    setActiveSteps();
+    setVersionTabs();
+  }
+
+  function setActiveSteps() {
+    let stepsToEnable;
+    let stepsToDisable;
+    let i = 0;
+
+    steps.forEach((step) => {
+      if (stepsToEnable === undefined) {
+        if (!productState.get(step)[0]) {
+          stepsToEnable = steps.slice(0, i + 1);
+          stepsToDisable = steps.slice(i + 1);
+        } else if (i < steps.length) {
+          i++;
+        }
+      }
+    });
+
+    if (stepsToEnable) {
+      enableSteps(stepsToEnable);
+    }
+
+    if (stepsToDisable) {
+      disableSteps(stepsToDisable);
+    }
+  }
+
+  function setVersionTabs() {
+    const versionTabs = form.querySelectorAll(
+      ".js-shop-step--version .p-tabs__link"
+    );
+    const version = productState.get("version")[0];
+    const quantity = productState.get("quantity")[0];
+
+    if (version === "#other") {
+      // disable the rest of the form
+      disableSteps(["support", "add"]);
+    } else if (quantity) {
+      // user has completed the form up to this
+      // point, enable the rest of the form
+      enableSteps(["support", "add"]);
+    }
+
+    versionTabs.forEach((tab) => {
+      const tabContentID = tab.getAttribute("href");
+      const target = form.querySelector(tabContentID);
+
+      if (!version && tab.getAttribute("aria-selected") === "true") {
+        // set initial state on load
+        productState.set("version", [tab.getAttribute("href")]);
+      } else if (tabContentID === version) {
+        tab.setAttribute("aria-selected", true);
+        target.classList.remove("u-hide");
+      } else {
+        tab.setAttribute("aria-selected", false);
+        target.classList.add("u-hide");
+      }
+    });
+  }
+
+  function updateCartLineItem() {
+    const cartName = form.querySelector(".js-shop-product-id");
+    const cartStep = form.querySelector(`${stepClassPrefix}add`);
+    const quantity = productState.get("quantity")[0];
+    const support = productState.get("support")[0];
+    const type = productState.get("type")[0];
+
+    const productString = `uai-${support}-${type} x ${quantity}`;
+
+    cartName.innerHTML = `Your selected product id is ${productString}`;
+
+    if (type && quantity && support) {
+      cartStep.classList.remove("u-hide");
+    } else {
+      cartStep.classList.add("u-hide");
+    }
+  }
+
+  function updateProductState(inputElement) {
+    productState.set(inputElement.name, [inputElement.value]);
+
+    updateCartLineItem();
```

Comment:
This should be run by the render function. Which, is called when state is mutated.